google/cloud/pubsub/internal/tracing_message_batch.h

```diff
@@ -44,40 +45,22 @@ class TracingMessageBatch : public MessageBatch {
       std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
           message_spans)
       : child_(std::move(child)), message_spans_(std::move(message_spans)) {}
-  // For testing only.
-  TracingMessageBatch(
-      std::unique_ptr<MessageBatch> child,
-      std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
-          message_spans,
-      std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
-          batch_sink_spans)
-      : child_(std::move(child)),
-        message_spans_(std::move(message_spans)),
-        batch_sink_spans_(std::move(batch_sink_spans)) {}
+
   ~TracingMessageBatch() override = default;
 
   void SaveMessage(pubsub::Message m) override;
 
-  void Flush() override;
-
-  void FlushCallback() override;
+  std::function<void(future<void>)> Flush() override;
 
   // For testing only.
   std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
   GetMessageSpans() const;
 
-  // For testing only.
-  std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
-  GetBatchSinkSpans() const;
-
  private:
   std::unique_ptr<MessageBatch> child_;
   std::mutex message_mu_;
-  std::mutex batch_sink_mu_;
```

Comment:
nit: there is a single `*mu_` in this class now.  You could go back to the more canonical `mu_` name.

Reply:
Done